### DISCUSSION TITLE
Change Apache Httpd Binary Provider

### DISCUSSION THREAD

**Discussion by mytlogos** [chocolatey-community/chocolatey-packages#2216] [General]
The package [https://community.chocolatey.org/packages/apache-httpd](https://community.chocolatey.org/packages/apache-httpd) is currently outdated.

It is based on the project [https://www.apachehaus.com](https://www.apachehaus.com/), but that project is currently on hold indefinitely, so the new release 2.4.56 is currently not available and not any newer ones. (Current latest of the package is 2.4.55).

Can you switch to another apache2 project for windows for the chocolatey package?

**Comment by pauby** [comment]
I think switching from one project to another maybe possible, but it all depends on the specifics / context. Is there another project that has taken this over?

**Reply by mytlogos** [reply]
The page https://httpd.apache.org/docs/current/platform/windows.html lists a few "popular" distributions:

- [ApacheHaus](http://www.apachehaus.com/cgi-bin/download.plx) (onhold, latest: 2.4.55)
- [Apache Lounge](http://www.apachelounge.com/download/) (latest: 2.4.57)
- [Bitnami WAMP Stack](http://bitnami.com/stack/wamp) (discontinued)
- [WampServer](http://www.wampserver.com/en/) (WAMP Stack only)
- [XAMPP](http://www.apachefriends.org/en/xampp.html) (Error 404)

In light of these findings, i would recommend Apache Lounge as an replacement.
But there may be ultimately other projects which provide binaries.

**Reply by pauby** [reply]
Are any / all of them a _direct_ replacement for `apache-httpd`?

My concern is that a user expects `apache-httpd` and if we simply switch to something else then that expectation is broken if it's not a direct replacement. If these are all just forks of the original project and are going to do their own thing, then I would suggest a new package is created for them.

**Reply by mytlogos** [reply]
I do not know of any "official" direct replacements for ApacheHause.
But a post on the [ApacheHaus forum](https://forum.apachehaus.com/announcements/apache-haus-project-is-on-hold/) says the following:

>  In the meantime, you can get updates from Apache Lounge. Their binaries are 100% compatible with ours. 

The same post also says that they do intend to return in the future.
So either we would use ApacheLounge for now in the same package and return back to ApacheHouse in the future, or create another package.

**Reply by pauby** [reply]
I can see the advantages and disadvantages to doing both. My gut tells me create a new package, as that manages user expectations in a better way.

@AdmiringWorm thoughts?

**Reply by AdmiringWorm** [reply]
It depends. Suppose both ApacheHous and ApacheLounge build their binary from the same source files in the same way, and they are just a different distributor and don't apply any of their own logic for the source files. In that case, we don't need a new package and can change our current package to use this other distributor.

However, if they make any changes or possibly will make any changes to the source code, then it will need to be a new package.

**Reply by pauby** [reply]
> However, if they make any changes or possibly will make any changes to the source code, then it will need to be a new package.

This is what concerns me. They are a new distribution so can set their own rules about what they want to do as there is no precedent. So, they may not simply do what was done before.

And in that case, the user experience is potentially bad.

**Reply by mytlogos** [reply]
And what now?
Should we ask ApacheLounge if they commit to stay compatible to ApacheHaus or similar?
Or just create a new package, as that is the "safest" way?

**Reply by pauby** [reply]
A new package sounds the safest way. I'd also add a Disqus comment to the original package page telling people there is a new package they may want to use.

**Reply by AdmiringWorm** [reply]
I agree with pauby here. The safest bet would be a new package.

Could you let us know if this is something you would like to work on (or if someone else comes across this discussion)?

If you are, we can create an issue based on the discussion here that any PR created needs to target.

**Reply by mytlogos** [reply]
Did you mean me or pauby?

**Reply by AdmiringWorm** [reply]
@mytlogos apologies, I meant you. If you are interested in creating a new package to provide the binaries from ApacheLounge in this repository, then do let us know so we can create an issue that any PR can be targeted against.

**Reply by mytlogos** [reply]
I am not really interested in creating a new package myself, i do not even know what is required to make a good package.
What will happen, if i would not do it?
How long would it roughly take for an up-to-date apache2 package?

**Reply by Starmania** [reply]
I open a discussion with a moderator as part of the inscription process of their forums. They seems to build the same source code in the same way as Apache haus, according to the mail I sent, it could potentially even come to discuss here.

**Reply by mytlogos** [reply]
Any update?
@Starmania , could you link the discussion?

**Reply by Starmania** [reply]
Oh, yeah sorry

**Reply by Starmania** [reply]
https://www.apachelounge.com/viewtopic.php?p=42009

**Reply by pauby** [reply]
@Starmania I've removed your previous post due to your use of language that is inappropriate and unnecessary here.

Please review the Code Of Conduct before posting further.

**Reply by JBlond** [reply]
Hi,
**tl;dr** The Apache Haus binaries can be replaced by Apache Lounge binaries.

---

I'm one of the owners of Apache Haus. For personal reasons, we can not do builds and provide updates.


We use the same source code and the same Visual Studio version and or the same SDK.
If not stated differently, our binaries are the same as Apache Lounge. e.g. we have a build with LibreSSL instead of OpenSSL.

Plus we offer some exotic 3rd party modules. 

if there are any questions left feel free to ask.

**Reply by mytlogos** [reply]
@AdmiringWorm and @pauby , are there still open questions for you for replacing ApacheHouse binaries with the ones of ApacheLounge?

Or do you still want a new package?

**Reply by mytlogos** [reply]
Any Updates?

**Reply by pauby** [reply]
Having looked at all of this, I'm going to swing towards a separate package, primarily out of caution. @AdmiringWorm ?

**Reply by AdmiringWorm** [reply]
Yeah, it's better to be safe than sorry. I would say to go with a new separate package as well.

**Reply by thkn-hofa** [reply]
@pauby any news on this? Before knowing about this issue, I started a pull request to change the source from ApacheHaus to ApacheLounge. It was of course denied but that was a few months ago. I'm not savvy enough to create a new package.

**Reply by pauby** [reply]
@thkn-hofa I'm unsure what news you are looking for?

The comments from myself and @AdmiringWorm above indicate that a separate package should be created for this.

**Reply by thkn-hofa** [reply]
It was my impression that you took ownership over making a new package. Guess I was wrong...

**Reply by pauby** [reply]
I didn't suggest I would create a new package. See [this comment](https://github.com/orgs/chocolatey-community/discussions/2216#discussioncomment-5807794)

**Reply by yakatz** [reply]
I would be happy to handle it.

**Comment by yakatz** [comment]
> Could you let us know if this is something you would like to work on (or if someone else comes across this discussion)?
> 
> If you are, we can create an issue based on the discussion here that any PR created needs to target.

I created a new package on my fork. Once there is an issue to link to, I can open a Pull Request.

https://github.com/yakatz/chocolatey-packages/tree/new-httpd-package

**Reply by yakatz** [reply]
https://github.com/chocolatey-community/chocolatey-packages/pull/2467
